### PR TITLE
Extract embedded class mapping into its own DTO

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

File: lib/Doctrine/ORM/Mapping/EmbeddedClassMapping.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+use ArrayAccess;
+
+use function property_exists;
+
+/** @template-implements ArrayAccess<string, mixed> */
+final class EmbeddedClassMapping implements ArrayAccess
+{
+    use ArrayAccessImplementation;
+
+    public string|bool|null $columnPrefix = null;
```
Comment (SenseException) [inline lib/Doctrine/ORM/Mapping/EmbeddedClassMapping.php]: Where does `bool` come from? From a PHP function that can return `false` on error and wasn't fixed yet?

**Review by SenseException — DISMISSED** [review]

File: lib/Doctrine/ORM/Mapping/EmbeddedClassMapping.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+     * @var class-string|null
+     */
+    public string|null $declared = null;
+
+    /** @param class-string $class */
+    public function __construct(public string $class)
+    {
+    }
+
+    /** @param array{class: class-string, columnPrefix: string|null, declaredField: string|null, originalField: string|null} $mappingArray */
```
Comment (derrabus) [inline lib/Doctrine/ORM/Mapping/EmbeddedClassMapping.php]:
```suggestion
    /** @param array{class: class-string, columnPrefix?: string|null, declaredField?: string|null, originalField?: string|null} $mappingArray */
```

File: tests/Doctrine/Tests/ORM/Mapping/EmbeddedClassMappingTest.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        $mapping->declared      = self::class; // sense
+
+        $resurrectedMapping = unserialize(serialize($mapping));
+        assert($resurrectedMapping instanceof EmbeddedClassMapping);
+
+        self::assertSame($resurrectedMapping->columnPrefix, 'these');
+        self::assertSame($resurrectedMapping->declaredField, 'values');
+        self::assertSame($resurrectedMapping->originalField, 'make');
+        self::assertSame($resurrectedMapping->inherited, self::class);
+        self::assertSame($resurrectedMapping->declared, self::class);
```
Comment (derrabus) [inline tests/Doctrine/Tests/ORM/Mapping/EmbeddedClassMappingTest.php]: Parameter order: expected value first, then the actual value.

**Review by derrabus — CHANGES_REQUESTED** [review]

File: lib/Doctrine/ORM/Mapping/EmbeddedClassMapping.php (hunk 1 above)
Comment (greg0ire) [reply lib/Doctrine/ORM/Mapping/EmbeddedClassMapping.php]: I must admit I didn't look into it until now. Apparently, it's the way to specify there should not be any column prefix: https://github.com/doctrine/orm/pull/1238. If you specify `null`, I think the column will be prefixed with the name of the entity the embedded class is embedded into.

Comment (greg0ire) [reply lib/Doctrine/ORM/Mapping/EmbeddedClassMapping.php]: Only `false` should be allowed, BTW.

Comment (greg0ire) [reply lib/Doctrine/ORM/Mapping/EmbeddedClassMapping.php]: Oh and it's legal in union types with php 8.1, so I can use it!

File: psalm-baseline.xml [hunk 4]
```diff
@@ -553,6 +554,7 @@
     </ArgumentTypeCoercion>
     <InvalidArgument>
       <code>$columnDef</code>
+      <code>$mapping</code>
```
Comment (greg0ire) [inline psalm-baseline.xml:557]: The new entries correspond to drivers (XMLDriver and AttributeDriver), when calling `mapEmbedded()`. An improvement could be to add some checks in the drivers regarding the types.

**Review by derrabus — APPROVED** [review]

**Pull request merged**